ui/src/containers/AppVersionsTableContainer.js

```diff
@@ -3,22 +3,55 @@ import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import { Checkbox, TextInput } from '@patternfly/react-core';
 import moment from 'moment';
-import { getApps, appDetailsSort, updateDisabledAppVersion, updateVersionCustomMessage } from '../actions/actions-ui';
+import { getApps, appDetailsSort } from '../actions/actions-ui';
 import AppsTable from '../components/AppsTable';
 import './TableContainer.css';
 import config from '../config/config';
 
 export class AppVersionsTableContainer extends React.Component {
+  shouldComponentUpdate = () => {
+    return this.state.shouldAppUpdate;
+  }
+  componentWillMount = () => {
+    this.setState({
+      ...this.state,
+      shouldAppUpdate: false
+    });
+  }
+  componentDidMount = () => {
+    this.setState({
+      ...this.state,
+      updatedVersions: this.props.appVersions
+    });
+  }
   handleDisableAppVersionChange = (_event, e) => {
     const id = e.target.id;
     const isDisabled = e.target.checked;
-    this.props.updateDisabledAppVersion(id, isDisabled);
+    this.state.updatedVersions.forEach(version => {
```

Comment:
This implementation will loop through every element in the array each time which seems wasteful give we're just changing on element - can we move towards an implementation more like the functional ones we have in our reducers when we return new state?